server/src/test/scala/org/http4s/netty/server/ServerTest.scala

```diff
@@ -117,6 +120,38 @@ abstract class ServerTest extends IOSuite {
       }
     }
   }
+
+  test("requests can be cancelled") {
+    val cancelled = Promise[Unit]()
+    val doCancel = IO.delay {
+      cancelled.success(())
+      ()
+    }
```

Comment:
This feels non-idiomatic. I'm happy to take suggestions to improve it.

Reply:
You can do something similar we do in DrainResponseTest with Defferred and IO.race

Reply:
Thanks, I'll take a look.

Reply:
That's a bit cleaner, thank you.